core/src/main/scala/org/http4s/websocket/WebSocket.scala

```diff
@@ -1,15 +1,23 @@
 package org.http4s.websocket
 
+import cats.effect.Concurrent
 import fs2._
 
-private[http4s] final case class WebSocket[F[_]](
+sealed trait WebSocket[F[_]] {
+  def receiveSend: Pipe[F, WebSocketFrame, WebSocketFrame]
+  def onClose: F[Unit]
+}
+
+private[http4s] final case class WebSocketSeparatePipe[F[_]: Concurrent](
     send: Stream[F, WebSocketFrame],
     receive: Pipe[F, WebSocketFrame, Unit],
     onClose: F[Unit]
-) {
-  @deprecated("Parameter has been renamed to `send`", "0.18.0-M7")
-  def read: Stream[F, WebSocketFrame] = send
-
-  @deprecated("Parameter has been renamed to `receive`", "0.18.0-M7")
-  def write: Pipe[F, WebSocketFrame, Unit] = receive
+) extends WebSocket[F] {
+  override def receiveSend: Pipe[F, WebSocketFrame, WebSocketFrame] =
```

Comment:
It seems like we could duplicate less of the builder functions by doing this in `build`, and then call `buildWithSinglePipe`?

Reply:
Made something of a currying with overloaded pair of `build` methods